lib/developer_portal/lib/liquid/forms/bot_protected.rb

```diff
@@ -27,14 +27,6 @@ def model
         @model ||= object.instance_variable_get(:@model)
       end
 
-      def bot_protection
-        protector = bot_protection_inputs
-
-        return '' if protector.blank?
-
-        form_builder.inputs { template.concat(protector) }
-      end
-
```

Comment:
This is very weird that this seems to not be needed :sweat_smile: 

But I'd say more - I can't seem to find a way to make `#model` and `#form_builder` called in this class either :upside_down_face: 
Removing them also doesn't seem to make a difference to me :thinking:  But maybe I'm missing something.

Reply:
Actually, removing `#model` probably makes sense because it was previously used to generate the bot protection inputs, but then this was removed from the model: https://github.com/3scale/porta/pull/3645/files#diff-01bcbc0f5b9247973c12e0adc67f8938f3c71b07fbac87bf10685f7bdd494c92L29

Reply:
As for `#form_builder` unfortunately there are some delegations to it:
```
delegate :input, :semantic_errors, to: :form_builder
```
But I am not sure when these are accessed :thinking:

Reply:
> Actually, removing `#model` probably makes sense because it was previously used to generate the bot protection inputs, but then this was removed from the model

> As for #form_builder unfortunately there are some delegations to it

And `#form_builder` calls `#model`, so we can't remove it either.

https://github.com/3scale/porta/blob/2436ac7d83cc98338719dc40183983649d306e4b/lib/developer_portal/lib/liquid/forms/bot_protected.rb#L32

Reply:
Yeah, but honestly, I can't find where these would be used :upside_down_face:

Reply:
Maybe we could add a TODO comment to investigate whether those could be removed...
Indeed probably it's safer to

Reply:
> Yeah, but honestly, I can't find where these would be used 🙃

Me neither.

> Maybe we could add a TODO comment to investigate whether those could be removed...

I don't like TODOs in general, because they are meant to be addressed later on, but we don't do it and they remain forever.

When somebody writes a TODO in a PR, I always comment: "Don't write a TODO, just fix it now". But I have too many open PRs right now and would like to spend my time on other priorities. So... what if we pretend we didn't see this? 😇

Reply:
okay :grimacing:

Reply:
why not remove these two methods. We don't concern models with bot protection anymore afaiu, so why keep these? If no tests fail I mean.

Reply:
We can't remove `#model` because it's called from `#form_builder`, and can't remove `#form_builder` because `:input` and `:semantic_errors` are delegated to it. We don't know where `:input` and `:semantic_errors` are being called from, and I don't rely on tests for this, even if tests pass we could find a problem in production and have to roll back. Just consider how many different developer portals we have in SaaS and on premises.

If you want to remove them in another PR, I'm fine, but I think it would require some investigation on why these methods exist and have a certain level of confidence of being safe to remove them.

Reply:
`semantic_errors` was for validation and IIRC we used the model errors to store them. But now we don't use a model so we can't query for errors. If anything uses the form_builder, wouldn't it fail given no model is used, regardless of the availability of the methods?

Reply:
`object.instance_variable_get(:@model)` will be always `nil` I assume.

Reply:
This discussion rings a :bell: :sweat_smile: 

https://github.com/3scale/porta/pull/3681#discussion_r1475896310